pkg/cmd/util/output/backup_describer.go

```diff
@@ -241,3 +247,98 @@ func failedDeletionCount(requests []v1.DeleteBackupRequest) int {
 	}
 	return count
 }
+
+// DescribePodVolumeBackups describes pod volume backups in human-readable format.
+func DescribePodVolumeBackups(d *Describer, backups []v1.PodVolumeBackup, details bool) {
+	if len(backups) == 0 {
+		d.Printf("Restic Backups: <none>\n")
+	} else {
+		if details {
+			d.Printf("Restic Backups:\n")
+		} else {
+			d.Printf("Restic Backups (specify --volume-details for more information):\n")
+		}
+
+		// separate backups by phase (combining <none>, New, InProgress into a single group)
+		backupsByPhase := groupByPhase(backups)
+
+		// go through phases in a specific order
+		for _, phase := range []v1.PodVolumeBackupPhase{
+			v1.PodVolumeBackupPhaseCompleted,
+			v1.PodVolumeBackupPhaseFailed,
+			"In Progress",
+			v1.PodVolumeBackupPhaseNew,
+		} {
+			if len(backupsByPhase[phase]) == 0 {
+				continue
+			}
+
+			// if we're not printing details, just report the phase and count
+			if !details {
+				d.Printf("\t%s:\t%d\n", phase, len(backupsByPhase[phase]))
+				continue
+			}
+
+			// within each phase grouping, group backups by pod
+			backupsByPod := make(map[string]*podVolumeBackupGroup)
+			for _, backup := range backupsByPhase[phase] {
+				key := fmt.Sprintf("%s/%s", backup.Spec.Pod.Namespace, backup.Spec.Pod.Name)
+
+				if group, ok := backupsByPod[key]; !ok {
+					backupsByPod[key] = &podVolumeBackupGroup{
+						namespace: backup.Spec.Pod.Namespace,
+						pod:       backup.Spec.Pod.Name,
+						volumes:   []string{backup.Spec.Volume},
+					}
+				} else {
+					group.volumes = append(group.volumes, backup.Spec.Volume)
+				}
+			}
+
+			// move all the groupings (grouped by pod) into a slice so we can sort it
+			var backupsByPodSlice []*podVolumeBackupGroup
+			for _, backups := range backupsByPod {
```

Comment:
Instead of looping here, could we add to the slice right after you create a new podVolumeBackupGroup and add it to the map in the loop above?

Reply:
Yes. Ended up extracting the map+slice into a separate struct to make the main logic simpler.